frontend-react/src/hooks/UseFilterManager.ts

```diff
@@ -0,0 +1,131 @@
+import React, { useMemo, useState } from "react";
+
+export type SortOrder = "ASC" | "DESC";
+export type PageSize = 10 | 25 | 50 | 100;
+export type StateUpdate<T> = React.Dispatch<React.SetStateAction<T>>;
+
```

Comment:
Why you're using a`React.Dispatch` here might need a comment. It's not a common thing to see.

Also, it seems to be mostly used like this:
```typescript
interface Store<S> {
    subscribe(listener: () => void): () => void;
    dispatch: React.Dispatch<ActionTypes>;
    getState(): S;
}
```